lib/src/realm_object.dart

```diff
@@ -267,6 +269,8 @@ mixin RealmObject on RealmEntity {
     final controller = RealmObjectNotificationsController<T>(object);
     return controller.createStream();
   }
+
+  late final DynamicRealmObject dynamic = DynamicRealmObject._(this);
```

Comment:
I'm not a super huge fan of hiding this API under a `dynamic` property like we did in .NET :/

Reply:
The API is kind of large and niche, so I'd rather not pollute the main interface but happy to discuss alternatives.